### PR TITLE
VMware: Add multiple IP Addresses in result

### LINKED ISSUE #34792: Return multiple IP addresses or add subnet filter in vmware facts
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

`vmware_vm_facts`

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 3c72161da5) last updated 2018/01/12 11:04:44 (GMT +200)
```

##### SUMMARY

In the current `devel` branch (at the time of writing), a single IP address is returned in the configuration summary. This is taken from the VMware vSphere API Python Bindings (`pyvmomi`).

If the VM has multiple interfaces (and therefore multiple IP addresses), only one of these addresses is returned. In such a case, the address that is returned might not be the address that you need.

For example, if you need the address of a specific interface to add to your in-memory inventory, you will need a list of all the IP addresses and then filter on the subnet in order to find and add the correct address to your inventory. SSH might not be accessible on all the interfaces from the host running the playbook due to networking, security and/or firewalling issues. Some interfaces might not be accessible externally at all.

What I propose is to add a property `ip_addresses` to the returned facts that include a list of all the IP addresses on all the interfaces. This list can then be iterated on and filtered down to the needed IP address using other modules. Alternatively, an extra argument can be added that can be used to pre-filter the list and return only the IP address in the given subnet.

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fix adds IPv4 and IPv6 addresses in virtual machine facts.

Fixes: #34792

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/vmware/vmware_vm_facts.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py [hunk 1]
```diff
@@ -55,61 +58,72 @@
...
+            if all_devices:
+                for dev in all_devices:
+                    if isinstance(dev, vim.vm.device.VirtualEthernetCard):
+                        _mac_address.append(dev.macAddress)
+
+            net_dict = {}
+            vmnet = _get_vm_prop(vm, ('guest', 'net'))
+            if vmnet:
+                for device in vmnet:
+                    import q
```
Comment (ghost) [inline lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: This is some debugging code that needs to be removed I believe.

**Comment by Akasurde** [comment]
@cvanniekerk Could you please take a look again ?

**Comment by ghost** [comment]
@Akasurde Yes that's better, thanks. Now you don't need the Python `q` package installed.

**Comment by Akasurde** [comment]
Yes. I installed `q` for my debugging stuff.

**Comment by Akasurde** [comment]
@cvanniekerk If it is working as per your requirement then I can merge it.

**Comment by ghost** [comment]
@Akasurde Yes please do!

**Comment by Akasurde** [comment]
@cvanniekerk Thanks for review and testing.

**Pull request merged**